Docs: No basic role documentation updates (#75110)
grafana/grafana: docs/sources/administration/roles-and-permissions/access-control/plan-rbac-rollout-strategy/index.md

## Review basic role and fixed role definitions

> As a first step in determining your permissions rollout strategy, we recommend that you become familiar with basic role and fixed role definitions. In addition to assigning fixed roles to any user and team, you can also modify basic roles permissions, which changes what a Viewer, Editor, or Admin can do. This flexibility means that there are many combinations of role assignments for you to consider. If you have a large number of Grafana users and teams, we recommend that you make a list of which fixed roles you might want to use. Keep in mind that `No Basic Role`, a role without any permissions, cannot be modified or updated.

Reviewer: ```suggestion
As a first step in determining your permissions rollout strategy, we recommend that you become familiar with basic role and fixed role definitions. In addition to assigning fixed roles to any user and team, you can also modify basic roles permissions, which changes what a Viewer, Editor, or Admin can do. This flexibility means that there are many combinations of role assignments for you to consider. If you have a large number of Grafana users and teams, we recommend that you make a list of which fixed roles you might want to use. Keep in mind that `No Basic Role`, which is a role without permissions, cannot be modified or updated.
```

Reviewer 2: I don't understand what you are trying to say @linoman.
Is it something along the line:
```suggestion
As a first step in determining your permissions rollout strategy, we recommend that you become familiar with basic role and fixed role definitions. In addition to assigning fixed roles to any user and team, you can also modify basic roles permissions, which changes what a Viewer, Editor, or Admin can do. This flexibility means that there are many combinations of role assignments for you to consider. If you have a large number of Grafana users and teams, we recommend that you make a list of which fixed roles you might want to use. Keep in mind that you can use the `No Basic Role`, which is a basic role without permissions that cannot be modified or updated.
```

```

- Or use [RBAC HTTP API]({{< relref "../../../../developers/http_api/access_control/#update-a-role" >}}).

### Manage user permissions through teams

> In the scenario where you want users to have granted access by the team they belong to, we recommend to set users role to `No Basic Role` and let the team assignment assign the role instead.

Reviewer 2: ```suggestion
In the scenario where you want users to have access granted by the team they belong to, we recommend to set users role to `No Basic Role` and let the team assignment assign the role instead.
```

1. In Grafana, ensure the following configuration settings are enabled.

>    1. `users.auto_assign_org = true` this will automatically assign users to the default organization.

Reviewer: How do these substeps render in the published docs? If they render as `1.`, I recommend manually changing these substeps to a., b., c. etc.

Author: I tried to render them with letters but the style was not the same as the numbers. I refactored it in order to show a configuration example instead.

Reviewer: Looks good! Thank you!

   1. `users.auto_assign_org_id = <org_id>` this will automatically assign users to the specified organization.
   1. `users.auto_assign_org_role = None` this will automatically assign users to the specified role.
   1. Restart the Grafana instance.

> 2. Create a team with the desired name.

Reviewer: these can be formatted as 1.

> 3. Set the fixed roles needed for the team.

Reviewer 2: ```suggestion
3. Assigned fixed roles to the team.
```

4. Add users to the team.

> This way, a user will be added automatically to the default organization and have no permissions until it has been assigned to a team.

Reviewer: ```suggestion
A user will be added to the default organization automatically but won't have any permissions until assigned to a team.
```

> ### Allow user to access only to a specific dashboard

Reviewer 2: I don't think I agree with this scenario and would not mention it here, at least yet.
I strongly feel like "No Basic Role" is an incorrect answer to the problem: "Viewers can access all dashboards by default".
IMHO, this use case is actually a feature request to: "Allow modifying the default dashboard/folder permissions".
Wdyt?

In the scenario where you want to allow a user access to a specific dashboard, we recommend the `No Basic Role` as a starting point for the user and then assign the dashboard permissions to the user.

> However, in order to access the dashboard more permissions are needed other than just the dashboard. Here's a list of permissions needed in order to access a dashboard:

Reviewer: ```suggestion
Below is a list of permissions required to access a dashboard:
```

Reviewer: I removed the "However..." sentence because it seems like we address that in line 326. Just checking that this makes sense to you and is ok.

> 1. The dashboard owner must give permission to the user

Reviewer: ```suggestion
1. The dashboard owner must give permission to the user.
```

1. The admin must grant the following permissions to the user

>    1. annotations:read

Reviewer: Do these permission need to be assigned in the order you have listed? If not, you can use a bulleted list here.

   1. datasource:read

> The permissions list here are the bare minimum needed to be able to view the dashboard. However, keep in mind that if the dashboard has multiple features, the permissions needed to see the dashboard might be more than just the ones listed here.

Reviewer: ```suggestion
Please note that the permissions listed below are the minimum required to access the dashboard. It's important to keep in mind that if the dashboard has multiple features, additional permissions may be required beyond those listed here.
```

> ### Reducing scope of service accounts

Reviewer: ```suggestion
### Reduce scope of service accounts
```

> Service Accounts are great tools for M2M communications. However, they are also a security risk if not properly scoped. To reduce the scope of a service account, you can create a Service Account with `No Basic Role` as a starting point and then assign the permissions needed for the service account.

Reviewer: ```suggestion
Using Service Accounts is an efficient way to facilitate M2M communications. However, they can pose a security threat if not scoped appropriately. To limit the scope of a service account, you can begin by creating a Service Account with "No Basic Role" and then assign the necessary permissions for the account.
```

> 1. Head over to the Service Accounts page and add a new Service Account.

Reviewer: ```suggestion
1. Refer to [Service Accounts]() and add a new Service Account.
```

Reviewer: Add a link to the doc.

1. Set the necessary roles for the Service Account.

>    1. Set the basic role to `No Basic Role`.

Reviewer: Same question here about how these substeps render. If they render as `1.`, then manually change these to a. and b.

Author: Same as before, I didn't like the final rendered style so I refactored it in order to keep some visual coherence.

   1. Set the fixed roles needed for the Service Account.

> This will reduce the permissions needed for the Service Account and reduce the risk of the Service Account being compromised.

Reviewer: ```suggestion
This will reduce the required permissions for the Service Account and minimize the risk of compromise.
```
